lib/NetPacksLib.cpp

```diff
@@ -1730,14 +1730,18 @@ DLL_LINKAGE std::string CatapultAttack::AttackInfo::toString() const
 					  % destinationTile % static_cast<int>(attackedPart) % static_cast<int>(damageDealt));
 }
 
-DLL_LINKAGE std::ostream & operator<<(std::ostream & out, const CatapultAttack::AttackInfo & attackInfo)
-{
-	return out << attackInfo.toString();
-}
-
 DLL_LINKAGE std::string CatapultAttack::toString() const
 {
-	return boost::str(boost::format("{CatapultAttack: attackedParts '%s', attacker '%d'}") % attackedParts % attacker);
+	std::ostringstream out;
+	out << "[";
+	for(auto it = attackedParts.begin(); it != attackedParts.end(); ++it)
+	{
+		out << it->toString();
+		if(std::prev(attackedParts.end()) != it) out << ", ";
+	}
+	out << "]";
+
+	return boost::str(boost::format("{CatapultAttack: attackedParts '%s', attacker '%d'}") % out.str() % attacker);

```

Comment:
this need more refactoring, for now it was moved from global.h 
`template<typename T>
std::ostream & operator<<(std::ostream & out, const std::vector<T> & container)`

I do not like default stream operation to be debug printing.